install-dependencies.sh

```diff
@@ -102,63 +103,102 @@ asio_not_packaged() {
    sudo apt install git cmake g++ gcc gettext libasio-dev libglew-dev libpng-dev libsdl2-dev \
     libsdl2-image-dev libsdl2-mixer-dev libsdl2-ttf-dev python zlib1g-dev
-elif [ "$DISTRO" == "solus" ]; then
-   echo "Installing dependencies for Solus..."
-   sudo eopkg install -c system.devel
-   sudo eopkg install git gettext glew-devel libicu-devel libpng-devel sdl2-devel sdl2-image-devel sdl2-mixer-devel sdl2-ttf-devel python
-   asio_not_packaged "Solus" "sudo "
+
 elif [ "$DISTRO" == "freebsd" ]; then
    echo "Installing dependencies for FreeBSD..."
    pkg install git asio cmake gettext glew png sdl2_image sdl2_mixer sdl2_net sdl2_ttf
+
 elif [ "$DISTRO" == "openbsd" ]; then
    echo "Installing dependencies for OpenBSD..."
-   pkg_add git cmake gcc g++ gettext-tools glew icu4c libexecinfo png \
+   doas pkg_add git cmake gcc g++ gettext-tools glew icu4c libexecinfo png \
     sdl2-image sdl2-mixer sdl2-net sdl2-ttf
-   asio_not_packaged "OpenBSD"
+   asio_not_packaged "OpenBSD" "doas"
+
 elif [ "$DISTRO" == "msys32" ]; then
    echo "Installing dependencies for 32-bit Windows..."
    pacman -S pacman -S mingw-w64-i686-toolchain git mingw-w64-i686-cmake \
     mingw-w64-i686-ninja mingw-w64-i686-asio mingw-w64-i686-SDL2_ttf \
     mingw-w64-i686-SDL2_mixer mingw-w64-i686-SDL2_image \
     mingw-w64-i686-glbinding mingw-w64-i686-glew mingw-w64-i686-icu
+
 elif [ "$DISTRO" == "msys64" ]; then
    echo "Installing dependencies for 64-bit Windows..."
    pacman -S mingw-w64-x86_64-toolchain git mingw-w64-x86_64-cmake \
     mingw-w64-x86_64-ninja mingw-w64-x86_64-asio mingw-w64-x86_64-SDL2_ttf \
     mingw-w64-x86_64-SDL2_mixer mingw-w64-x86_64-SDL2_image \
     mingw-w64-x86_64-glbinding mingw-w64-x86_64-glew mingw-w64-i686-icu
+
 elif [ "$DISTRO" == "homebrew" ]; then
    echo "Installing dependencies for Mac Homebrew..."
    brew install asio git cmake doxygen gettext glew graphviz icu4c jpeg \
     libogg libpng libvorbis ninja python sdl2 sdl2_image sdl2_mixer sdl2_ttf zlib
+
+elif [ "$DISTRO" == "solus" ]; then
+   echo "Installing dependencies for Solus..."
+   sudo eopkg install -c system.devel
+   sudo eopkg install git gettext glew-devel libicu-devel libpng-devel sdl2-devel \
+    sdl2-image-devel sdl2-mixer-devel sdl2-ttf-devel python
+
+   if ! [ -f /usr/include/asio.hpp -o -f ./auto_dependencies/asio/asio.hpp ]
+   then
+     echo "Installing Asio from source..."
+     rm -r ./auto_dependencies/asio >/dev/null 2>/dev/null || true
+     mkdir -p ./auto_dependencies/asio
+     cd ./auto_dependencies/asio
+
+     if which wget > /dev/null
+     then
+       REMOTE="wget -O"
+     elif which curl > /dev/null
+     then
+       REMOTE="curl -o"
+     else
+       sudo eopkg install wget
+       REMOTE="wget -O"
+     fi
+
+     $REMOTE sources https://sourceforge.net/projects/asio/files/latest/download
+     unzip sources
+     mv asio* extracted
+     mv extracted/include/asio* .
+     rm -r extracted sources
```

Comment:
If, after @frankystone's very valid point, we still decide that maybe we can still use a fixed version (with a hash check after downloading) and accept the responsibility of not being able to track its security updates, we should download the `.tar.bz2` instead of the `.zip` (1.8MB instead of 8.7MB too) and extract it with
```
tar --strip-components=2 --wildcards --exclude 'Makefile*' -xjf asio-<ver>.tar.bz2 '*/include/*'
```